File: webview/app/src/main/java/io/github/michaelbui99/webview/MainActivity.java
package io.github.michaelbui99.webview;

import androidx.appcompat.app.AppCompatActivity;

import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private WebView webView;
    private EditText urlInput;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button goButton = findViewById(R.id.button_go);
        webView = findViewById(R.id.web_view);
        urlInput = findViewById(R.id.input_url);

        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new WebViewClient());
        webView.loadUrl("https://google.com");
        goButton.setOnClickListener(this::onGo);
    }


    public void onGo(View v) {
        String url ="";
        if (!urlInput.getText().toString().contains("https://")){
            url = "https://" + urlInput.getText().toString();
        }else{
            url = urlInput.getText().toString();
        }
        webView.loadUrl(url);
        urlInput.setText(webView.getUrl());
    }
}

File: PersistenceRoom/app/src/main/java/io/github/michaelbui99/persistenceroom/MainActivity.kt
package io.github.michaelbui99.persistenceroom

import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.lifecycle.ViewModelProvider
import io.github.michaelbui99.persistenceroom.model.Note
import io.github.michaelbui99.persistenceroom.ui.MainViewModel

class MainActivity : AppCompatActivity() {
    private lateinit var titleInput: EditText
    private lateinit var addNoteButton: Button
    private lateinit var deleteAllButton: Button
    private lateinit var noteView: TextView
    private lateinit var viewModel: MainViewModel


    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewModel = ViewModelProvider(this)[MainViewModel::class.java]

        titleInput = findViewById(R.id.input_title)
        addNoteButton = findViewById(R.id.button_addNote)
        deleteAllButton = findViewById(R.id.button_deleteAll)
        noteView = findViewById(R.id.textview_notes)

        viewModel.getAllNotes().observe(this) {
            renderNotes(it)
        }

        addNoteButton.setOnClickListener() {
            viewModel.addNote(titleInput.text.toString())
            titleInput.setText("")
        }

        deleteAllButton.setOnClickListener(){
            viewModel.deleteAllNotes()
        }
    }

    private fun renderNotes(notes: List<Note>) {
        noteView.text = ""
        notes.forEach {
            noteView.append("${it.title}\n")
        }
    }
}